nav2_costmap_2d/src/costmap_topic_collision_checker.cpp

```diff
@@ -107,68 +99,7 @@ Footprint CollisionChecker::getFootprint(const geometry_msgs::msg::Pose2D & pose
   return footprint;
 }
 
-double CollisionChecker::footprintCost(const Footprint footprint)
-{
-  // now we really have to lay down the footprint in the costmap_ grid
-  unsigned int x0, x1, y0, y1;
-  double footprint_cost = 0.0;
-
-  // we need to rasterize each line in the footprint
-  for (unsigned int i = 0; i < footprint.size() - 1; ++i) {
-    // get the cell coord of the first point
-    worldToMap(footprint[i].x, footprint[i].y, x0, y0);
-
-    // get the cell coord of the second point
-    worldToMap(footprint[i + 1].x, footprint[i + 1].y, x1, y1);
-
-    footprint_cost = std::max(lineCost(x0, x1, y0, y1), footprint_cost);
-  }
-
-  // we also need to connect the first point in the footprint to the last point
-  // get the cell coord of the last point
-  worldToMap(footprint.back().x, footprint.back().y, x0, y0);
-
-  // get the cell coord of the first point
-  worldToMap(footprint.front().x, footprint.front().y, x1, y1);
-
-  footprint_cost = std::max(lineCost(x0, x1, y0, y1), footprint_cost);
-
-  // if all line costs are legal... then we can return that the footprint is legal
-  return footprint_cost;
-}
-
-double CollisionChecker::lineCost(int x0, int x1, int y0, int y1) const
-{
-  double line_cost = 0.0;
-  double point_cost = -1.0;
-
-  for (nav2_util::LineIterator line(x0, y0, x1, y1); line.isValid(); line.advance()) {
-    point_cost = pointCost(line.getX(), line.getY());   // Score the current point
-
-    if (line_cost < point_cost) {
-      line_cost = point_cost;
-    }
-  }
-
-  return line_cost;
-}
-
-double CollisionChecker::pointCost(int x, int y) const
-{
-  unsigned char cost = costmap_->getCost(x, y);
-  // if the cell is in an obstacle the path is invalid or unknown
-  if (cost == LETHAL_OBSTACLE) {
-    RCLCPP_DEBUG(rclcpp::get_logger(name_), "Map Cell: [%d, %d]", x, y);
-    throw IllegalPoseException(name_, "Footprint Hits Obstacle.");
-  } else if (cost == NO_INFORMATION) {
-    RCLCPP_DEBUG(rclcpp::get_logger(name_), "Map Cell: [%d, %d]", x, y);
-    throw IllegalPoseException(name_, "Footprint Hits Unknown Region.");
-  }
-
-  return cost;
-}
-
-void CollisionChecker::unorientFootprint(
+void CostmapTopicCollisionChecker::unorientFootprint(
```

Comment:
What's this unorient footprint for?

Reply:
It is used to get normalized footprint from the positional footprint. You get positional footprint when you subscribe to the footprint topic. This class first normalizes the footprint from the subscription and then pushes it to the obstacle checking mechanism.

One thing is strange, this whole thing is first transform the coordinated footprint to normalized footprint and then transform it back to coordinated footprint using current coordinate of robot.

Let me check if we can eliminate the double transformation thing.

Reply:
I tried removing the `unorientedfootprint` and all the test do pass without it. Please take a look. At first glance this function seem redundant but might be a hidden secret behind its existence. Please review it and let me know.

Reply:
@shivaang12 it is used in the function above in master, it looks like you commented it out just 10 lines above. I think that's a regression.